tokio/src/io/util/empty.rs

```diff
@@ -71,6 +88,27 @@ impl AsyncBufRead for Empty {
     fn consume(self: Pin<&mut Self>, _: usize) {}
 }
 
+impl AsyncWrite for Empty {
```

Comment:
There are some optional methods you can implement: `poll_write_vectored`, `is_write_vectored`.

Reply:
Do you have a suggestion on how to implement [poll_write_vectored](https://docs.rs/tokio/latest/tokio/io/trait.AsyncWrite.html#method.poll_write_vectored)? It takes a slice of [IoSlice](https://doc.rust-lang.org/nightly/std/io/struct.IoSlice.html) which doesn't have a length as far as I can tell. I'm thinking we'll have to advance them to return the total number of bytes "written".

Reply:
Oh, nevermind. Apparently `len()` does work 😅